## **Introduction to Data Science Class**
### **Project 2, Part 1: Introduction to Data Analysis** <br>
### ***"The Impact of College Institutions on Career Salaries using Data Analysis"*** <br>

**Raven Shaw** <br>
**Student ID: 003099620** <br>
**Friday, April 4, 2025**

#### **Introduction and Background:**

This dataset examines the impact of college institutions on students' future career earnings. It includes a diverse selection of U.S. colleges, categorized by region and school type, along with salary data at various career stages—from median starting salary to the 90th percentile mid-career salary. By analyzing this data, we can gain insights into how different educational institutions influence long-term earning potential. <br>

The dataset originates from Kaggle, and I combined two of the datasets (regions and school type) to make it one cohesive dataset. I chose it because, as a college student, I often wonder how my own educational experience will shape my future career prospects. Beyond exploring the general impact of college institutions on earnings, I was particularly interested in comparing salaries across different schools to uncover potential wage disparities. Such disparities could highlight broader issues of income inequality rooted in systemic differences between institutions, such as variations in funding, resources, and networking opportunities. <br>

**"College_Salary" Dataset**: [Click here to view the orginal dataset source](https://www.kaggle.com/datasets/wsj/college-salaries?resource=download) <br>

#### **Section 1: Dataset Overview and Strcuture**

In [354]:
import numpy as np #importing Python libraries
import pandas as pd

In [355]:
df = pd.read_csv("College_Salary.csv") #importing the dataset

In [356]:
df #displaying the data

,School Name,Region,School Type,Starting Median Salary,Mid-Career Median Salary,Mid-Career 10th Percentile Salary,Mid-Career 25th Percentile Salary,Mid-Career 75th Percentile Salary,Mid-Career 90th Percentile Salary
0,Amherst College,Northeastern,Liberal Arts,54500.0,107000.0,NaN,84900.0,162000.0,NaN
1,Appalachian State University,Southern,State,40400.0,69100.0,37200.0,50400.0,90800.0,115000.0
2,Arizona State University (ASU),NaN,State,47400.0,84100.0,44600.0,60700.0,114000.0,163000.0
3,Arkansas State University (ASU),Southern,State,38700.0,63300.0,33600.0,45300.0,83900.0,118000.0
4,Auburn University,Southern,NaN,45400.0,84700.0,45400.0,62700.0,109000.0,145000.0
...,...,...,...,...,...,...,...,...,...
252,Whitman College,Western,Liberal Arts,43500.0,80100.0,NaN,64800.0,111000.0,NaN
253,Williams College,Northeastern,Liberal Arts,51700.0,102000.0,NaN,76400.0,143000.0,NaN
254,Wittenberg University,Midwestern,Liberal Arts,39200.0,78200.0,NaN,54100.0,131000.0,NaN
255,Worcester Polytechnic Institute (WPI),Northeastern,Engineering,61000.0,114000.0,80000.0,91200.0,137000.0,180000.0


In [357]:
print(df.head()) #show first five rows

                       School Name        Region   School Type  \
0                  Amherst College  Northeastern  Liberal Arts   
1     Appalachian State University      Southern         State   
2   Arizona State University (ASU)           NaN         State   
3  Arkansas State University (ASU)      Southern         State   
4                Auburn University      Southern           NaN   

   Starting Median Salary  Mid-Career Median Salary  \
0                 54500.0                  107000.0   
1                 40400.0                   69100.0   
2                 47400.0                   84100.0   
3                 38700.0                   63300.0   
4                 45400.0                   84700.0   

   Mid-Career 10th Percentile Salary  Mid-Career 25th Percentile Salary  \
0                                NaN                            84900.0   
1                            37200.0                            50400.0   
2                            44600.0           

In [358]:
print(df.tail()) #show last five rows

                               School Name        Region   School Type  \
252                        Whitman College       Western  Liberal Arts   
253                       Williams College  Northeastern  Liberal Arts   
254                  Wittenberg University    Midwestern  Liberal Arts   
255  Worcester Polytechnic Institute (WPI)  Northeastern   Engineering   
256                        Yale University  Northeastern    Ivy League   

     Starting Median Salary  Mid-Career Median Salary  \
252                 43500.0                   80100.0   
253                 51700.0                  102000.0   
254                 39200.0                   78200.0   
255                 61000.0                  114000.0   
256                 59100.0                  126000.0   

     Mid-Career 10th Percentile Salary  Mid-Career 25th Percentile Salary  \
252                                NaN                            64800.0   
253                                NaN                    

In [359]:
df.info #displaying general information about the data

<bound method DataFrame.info of                                School Name        Region   School Type  \
0                          Amherst College  Northeastern  Liberal Arts   
1             Appalachian State University      Southern         State   
2           Arizona State University (ASU)           NaN         State   
3          Arkansas State University (ASU)      Southern         State   
4                        Auburn University      Southern           NaN   
..                                     ...           ...           ...   
252                        Whitman College       Western  Liberal Arts   
253                       Williams College  Northeastern  Liberal Arts   
254                  Wittenberg University    Midwestern  Liberal Arts   
255  Worcester Polytechnic Institute (WPI)  Northeastern   Engineering   
256                        Yale University  Northeastern    Ivy League   

     Starting Median Salary  Mid-Career Median Salary  \
0                   54

In [360]:
df.shape #see number of rows and columns (respectively)

(257, 9)

In [361]:
df.columns #displaying only column names

Index(['School Name', 'Region', 'School Type', 'Starting Median Salary',
       'Mid-Career Median Salary', 'Mid-Career 10th Percentile Salary',
       'Mid-Career 25th Percentile Salary',
       'Mid-Career 75th Percentile Salary',
       'Mid-Career 90th Percentile Salary'],
      dtype='object')

In [362]:
print(df.dtypes) #viewing datatypes

School Name                           object
Region                                object
School Type                           object
Starting Median Salary               float64
Mid-Career Median Salary             float64
Mid-Career 10th Percentile Salary    float64
Mid-Career 25th Percentile Salary    float64
Mid-Career 75th Percentile Salary    float64
Mid-Career 90th Percentile Salary    float64
dtype: object


**Based on the datatypes listed above the variable types for each column are:**

* 'School Name' = categorical
* 'Region' = Categorical
* 'School Type' = Categorical
* 'Starting Median Salary' = numerical
* 'Mid-Career Median Salary' = numerical
* 'Mid-Career 10th Percentile Salary' = numerical
* 'Mid-Career 25th Percentile Salary' = numerical
* 'Mid-Career 75th Percentile Salary' = numerical
* 'Mid-Career 90th Percentile Salary' = numerical



In [364]:
print(df.isnull().sum()) #Checking for null values

School Name                           0
Region                                5
School Type                           4
Starting Median Salary                0
Mid-Career Median Salary              0
Mid-Career 10th Percentile Salary    45
Mid-Career 25th Percentile Salary     0
Mid-Career 75th Percentile Salary     0
Mid-Career 90th Percentile Salary    45
dtype: int64


In [365]:
# Calculate the percentage of missing values for each column
missing_data = df.isnull().sum()/len(df)*100 #len refers to "length" which is number of rows
print(missing_data) #printing null values in percentage form

School Name                           0.000000
Region                                1.945525
School Type                           1.556420
Starting Median Salary                0.000000
Mid-Career Median Salary              0.000000
Mid-Career 10th Percentile Salary    17.509728
Mid-Career 25th Percentile Salary     0.000000
Mid-Career 75th Percentile Salary     0.000000
Mid-Career 90th Percentile Salary    17.509728
dtype: float64


**Plan to Handle Missing Values**: To handle the missing values in the "region columns" I plan to fill in using "unknown" since it's categorical yet systematic data. However, for the "school type" column, I plan to use mode imputations since it'll help retain data. For the 10th and 90th percentile missing values, I will fill in using the median. I don't want to drop these columns because insight into these data points can help highlight potential income inequality and earning disparities across and within schools. Additionally, I'm using the median rather than the mean because the 10th and 90th percentiles are normally outliers and have extreme values therefore the median will be mroe robust and less influenced by these extreme values.

In [367]:
df_new=df.copy() #creating copy of the dataframe to ensure original dataframe isn't impacted

***Section 1 Explanation***: I began by importing the libraries "numpy" and "pandas" and setting them equal to the variables "np" and "pd" respectively. I then imported my csv (ensuring it was downloaded into the correct folder on my desktop) by using the "read_csv" function and setting the dataset equal to the variable "df." I then sought to gain an overview of my data by using various functions such as "describe", "head", "tail," "type," "shape," and "columns." This allowed me to understand my dataset more clearly. From there I checked the number of null values and then calculated the percentage of those missing values within the entire dataset to see how much of an impact those missing values have on the overall datset.

#### **Section 2: Data Cleaning and Indexing**

In [370]:
df_new.duplicated().sum() #checking for duplicates

5

In [371]:
df_drop_dup=df_new.copy()
df_drop_dup.drop_duplicates(inplace=True) #now only have 252 rows demonstrating that the duplicated rows were dropped

In [372]:
df_drop_dup.fillna({'Mid-Career 10th Percentile Salary': df_new['Mid-Career 10th Percentile Salary'].median(),
               'Mid-Career 90th Percentile Salary': df_new['Mid-Career 90th Percentile Salary'].median()}, inplace=True) 
#filling in missing values in both 10th and 90th percentile columns with median; using inplace=True to ensure changes stick
df_drop_dup #displaying edited data 

,School Name,Region,School Type,Starting Median Salary,Mid-Career Median Salary,Mid-Career 10th Percentile Salary,Mid-Career 25th Percentile Salary,Mid-Career 75th Percentile Salary,Mid-Career 90th Percentile Salary
0,Amherst College,Northeastern,Liberal Arts,54500.0,107000.0,43150.0,84900.0,162000.0,151000.0
1,Appalachian State University,Southern,State,40400.0,69100.0,37200.0,50400.0,90800.0,115000.0
2,Arizona State University (ASU),NaN,State,47400.0,84100.0,44600.0,60700.0,114000.0,163000.0
3,Arkansas State University (ASU),Southern,State,38700.0,63300.0,33600.0,45300.0,83900.0,118000.0
4,Auburn University,Southern,NaN,45400.0,84700.0,45400.0,62700.0,109000.0,145000.0
...,...,...,...,...,...,...,...,...,...
252,Whitman College,Western,Liberal Arts,43500.0,80100.0,43150.0,64800.0,111000.0,151000.0
253,Williams College,Northeastern,Liberal Arts,51700.0,102000.0,43150.0,76400.0,143000.0,151000.0
254,Wittenberg University,Midwestern,Liberal Arts,39200.0,78200.0,43150.0,54100.0,131000.0,151000.0
255,Worcester Polytechnic Institute (WPI),Northeastern,Engineering,61000.0,114000.0,80000.0,91200.0,137000.0,180000.0


In [373]:
df_region_type = df_drop_dup.copy() #equating new copy to "df_region_type"
type_mode = df_region_type['School Type'].mode()[0] #selecting most frequent category
df_region_type['School Type']=df_region_type['School Type'].fillna(type_mode)  #using mode imputation for "School Type" values 
df_region_type #printing changes

,School Name,Region,School Type,Starting Median Salary,Mid-Career Median Salary,Mid-Career 10th Percentile Salary,Mid-Career 25th Percentile Salary,Mid-Career 75th Percentile Salary,Mid-Career 90th Percentile Salary
0,Amherst College,Northeastern,Liberal Arts,54500.0,107000.0,43150.0,84900.0,162000.0,151000.0
1,Appalachian State University,Southern,State,40400.0,69100.0,37200.0,50400.0,90800.0,115000.0
2,Arizona State University (ASU),NaN,State,47400.0,84100.0,44600.0,60700.0,114000.0,163000.0
3,Arkansas State University (ASU),Southern,State,38700.0,63300.0,33600.0,45300.0,83900.0,118000.0
4,Auburn University,Southern,State,45400.0,84700.0,45400.0,62700.0,109000.0,145000.0
...,...,...,...,...,...,...,...,...,...
252,Whitman College,Western,Liberal Arts,43500.0,80100.0,43150.0,64800.0,111000.0,151000.0
253,Williams College,Northeastern,Liberal Arts,51700.0,102000.0,43150.0,76400.0,143000.0,151000.0
254,Wittenberg University,Midwestern,Liberal Arts,39200.0,78200.0,43150.0,54100.0,131000.0,151000.0
255,Worcester Polytechnic Institute (WPI),Northeastern,Engineering,61000.0,114000.0,80000.0,91200.0,137000.0,180000.0


In [374]:
df_region_type['Region']=df_region_type['Region'].fillna('Unknown') #replacing missing values in region with "Unknown"
df_region_type #displaying the data

,School Name,Region,School Type,Starting Median Salary,Mid-Career Median Salary,Mid-Career 10th Percentile Salary,Mid-Career 25th Percentile Salary,Mid-Career 75th Percentile Salary,Mid-Career 90th Percentile Salary
0,Amherst College,Northeastern,Liberal Arts,54500.0,107000.0,43150.0,84900.0,162000.0,151000.0
1,Appalachian State University,Southern,State,40400.0,69100.0,37200.0,50400.0,90800.0,115000.0
2,Arizona State University (ASU),Unknown,State,47400.0,84100.0,44600.0,60700.0,114000.0,163000.0
3,Arkansas State University (ASU),Southern,State,38700.0,63300.0,33600.0,45300.0,83900.0,118000.0
4,Auburn University,Southern,State,45400.0,84700.0,45400.0,62700.0,109000.0,145000.0
...,...,...,...,...,...,...,...,...,...
252,Whitman College,Western,Liberal Arts,43500.0,80100.0,43150.0,64800.0,111000.0,151000.0
253,Williams College,Northeastern,Liberal Arts,51700.0,102000.0,43150.0,76400.0,143000.0,151000.0
254,Wittenberg University,Midwestern,Liberal Arts,39200.0,78200.0,43150.0,54100.0,131000.0,151000.0
255,Worcester Polytechnic Institute (WPI),Northeastern,Engineering,61000.0,114000.0,80000.0,91200.0,137000.0,180000.0


In [375]:
df_region_type.isnull().sum() #rechecking the null values

School Name                          0
Region                               0
School Type                          0
Starting Median Salary               0
Mid-Career Median Salary             0
Mid-Career 10th Percentile Salary    0
Mid-Career 25th Percentile Salary    0
Mid-Career 75th Percentile Salary    0
Mid-Career 90th Percentile Salary    0
dtype: int64

***Section 2 Explanation***: For this section, I focused on replacing missing values. I started by removing the 5 duplicates. As aforementioned in section 1, I replaced the missing values in the "region," "school type," and "10th and 90th percentile columns." Nevertheless, I used different methods for each. For the region I replaced using the word "Unknown" because region is systematic data whereas for school type I could rely on mode. For the numerical columns, I focused on replacing missing values using median. I then rechecked the null values to ensure they are all fill in. I then saved the cleaned dataset (see below).

#### **Section 3: Descriptive Statistics**

In [378]:
df_cleaned = df_region_type.to_csv('college_salary_updated.csv', index=False) 
#saving cleaned dataset; removing extra column using 'index=False'

In [379]:
df_cleaned = pd.read_csv('college_salary_updated.csv') #reading updated dataset
df_cleaned #displaying cleaned dataset

,School Name,Region,School Type,Starting Median Salary,Mid-Career Median Salary,Mid-Career 10th Percentile Salary,Mid-Career 25th Percentile Salary,Mid-Career 75th Percentile Salary,Mid-Career 90th Percentile Salary
0,Amherst College,Northeastern,Liberal Arts,54500.0,107000.0,43150.0,84900.0,162000.0,151000.0
1,Appalachian State University,Southern,State,40400.0,69100.0,37200.0,50400.0,90800.0,115000.0
2,Arizona State University (ASU),Unknown,State,47400.0,84100.0,44600.0,60700.0,114000.0,163000.0
3,Arkansas State University (ASU),Southern,State,38700.0,63300.0,33600.0,45300.0,83900.0,118000.0
4,Auburn University,Southern,State,45400.0,84700.0,45400.0,62700.0,109000.0,145000.0
...,...,...,...,...,...,...,...,...,...
247,Whitman College,Western,Liberal Arts,43500.0,80100.0,43150.0,64800.0,111000.0,151000.0
248,Williams College,Northeastern,Liberal Arts,51700.0,102000.0,43150.0,76400.0,143000.0,151000.0
249,Wittenberg University,Midwestern,Liberal Arts,39200.0,78200.0,43150.0,54100.0,131000.0,151000.0
250,Worcester Polytechnic Institute (WPI),Northeastern,Engineering,61000.0,114000.0,80000.0,91200.0,137000.0,180000.0


In [380]:
df_cleaned.describe() #provides general statistics

,Starting Median Salary,Mid-Career Median Salary,Mid-Career 10th Percentile Salary,Mid-Career 25th Percentile Salary,Mid-Career 75th Percentile Salary,Mid-Career 90th Percentile Salary
count,252.000000,252.000000,252.000000,252.000000,252.000000,252.000000
mean,46088.888889,84003.968254,44516.071429,60496.031746,116365.079365,157371.031746
std,6578.084128,14845.348317,8046.486554,11768.506835,23811.322820,32689.316123
min,34800.000000,43900.000000,25600.000000,31800.000000,60900.000000,91300.000000
25%,42000.000000,73400.000000,39775.000000,53000.000000,99975.000000,138000.000000
50%,44800.000000,81600.000000,43150.000000,58500.000000,112000.000000,151000.000000
75%,48150.000000,92850.000000,46400.000000,65700.000000,128000.000000,165000.000000
max,75500.000000,134000.000000,80000.000000,104000.000000,234000.000000,326000.000000


*Interesting Takeaway*: There is a disproportionately large difference between the Mid-Career 90th Percentile group mean and Mid-Career 10th Percentile group mean which is something I want to explore further using descriptive analysis.

In [382]:
df_new["Region"].value_counts() #analyzing specific data points in region column

Region
Northeastern    70
Midwestern      63
Southern        57
Western         35
California      27
Name: count, dtype: int64

In [383]:
df_new["School Type"].value_counts() #analyzing specific data points in school type column

School Type
State           160
Liberal Arts     49
Engineering      19
Party            16
Ivy League        9
Name: count, dtype: int64

In [384]:
salary_by_type1 = df_cleaned.groupby('School Type')['Starting Median Salary'].median() 
# calculating the median starting salary for each school type in the cleaned dataset
salary_by_type1

School Type
Engineering     51650.0
Ivy League      59250.0
Liberal Arts    45400.0
Party           44400.0
State           43800.0
Name: Starting Median Salary, dtype: float64

In [385]:
salary_by_type2 = df_cleaned.groupby('School Type')['Mid-Career Median Salary'].median()
# calculating the median mid career salary for each school type in the cleaned dataset
salary_by_type2

School Type
Engineering      98850.0
Ivy League      115000.0
Liberal Arts     83900.0
Party            81650.0
State            78900.0
Name: Mid-Career Median Salary, dtype: float64

In [386]:
salary_by_region1 = df_cleaned.groupby('Region')['Mid-Career 10th Percentile Salary'].mean()
#calculating the mean salary for each region in the 10th percentile
salary_by_region1

Region
California      45842.307692
Midwestern      42528.225806
Northeastern    48610.294118
Southern        42492.105263
Unknown         42970.000000
Western         42558.823529
Name: Mid-Career 10th Percentile Salary, dtype: float64

In [387]:
salary_by_region2 = df_cleaned.groupby('Region')['Mid-Career 90th Percentile Salary'].mean()
#calculating the mean salary for each region in the 90th percentile
salary_by_region2

Region
California      159307.692308
Midwestern      146967.741935
Northeastern    180985.294118
Southern        147219.298246
Unknown         156600.000000
Western         144764.705882
Name: Mid-Career 90th Percentile Salary, dtype: float64

In [388]:
salary_inequality = salary_by_region2 - salary_by_region1 #subtracting the difference between mean values of the 90th and 10th percentiles
salary_inequality

Region
California      113465.384615
Midwestern      104439.516129
Northeastern    132375.000000
Southern        104727.192982
Unknown         113630.000000
Western         102205.882353
dtype: float64

In [389]:
def classify_status(salary): #feature engineering beginning with creating a user defined function
    if pd.isnull(salary):
        return 'Unknown'
    elif salary < 30000:
        return 'Lower Class'
    elif salary < 94000:
        return 'Middle Class'
    elif salary < 153000:
        return 'Upper Middle Class'
    else:
        return 'Upper CLass'
df_cleaned=df_cleaned.copy()
df_cleaned['Median Socioeconomic Class']=df_cleaned['Mid-Career Median Salary'].apply(classify_status)
#creating a new column "Median Socioeconomic Class" based on defined function above
df_cleaned

,School Name,Region,School Type,Starting Median Salary,Mid-Career Median Salary,Mid-Career 10th Percentile Salary,Mid-Career 25th Percentile Salary,Mid-Career 75th Percentile Salary,Mid-Career 90th Percentile Salary,Median Socioeconomic Class
0,Amherst College,Northeastern,Liberal Arts,54500.0,107000.0,43150.0,84900.0,162000.0,151000.0,Upper Middle Class
1,Appalachian State University,Southern,State,40400.0,69100.0,37200.0,50400.0,90800.0,115000.0,Middle Class
2,Arizona State University (ASU),Unknown,State,47400.0,84100.0,44600.0,60700.0,114000.0,163000.0,Middle Class
3,Arkansas State University (ASU),Southern,State,38700.0,63300.0,33600.0,45300.0,83900.0,118000.0,Middle Class
4,Auburn University,Southern,State,45400.0,84700.0,45400.0,62700.0,109000.0,145000.0,Middle Class
...,...,...,...,...,...,...,...,...,...,...
247,Whitman College,Western,Liberal Arts,43500.0,80100.0,43150.0,64800.0,111000.0,151000.0,Middle Class
248,Williams College,Northeastern,Liberal Arts,51700.0,102000.0,43150.0,76400.0,143000.0,151000.0,Upper Middle Class
249,Wittenberg University,Midwestern,Liberal Arts,39200.0,78200.0,43150.0,54100.0,131000.0,151000.0,Middle Class
250,Worcester Polytechnic Institute (WPI),Northeastern,Engineering,61000.0,114000.0,80000.0,91200.0,137000.0,180000.0,Upper Middle Class


In [428]:
df_cleaned.to_csv('Raven_Shaw_project2_part1_data_cleaned.csv') #saving the cleaned CSV

***Section 3 Explanation***: This section was based on descriptive statistics. After revieiwng the general statistics (based on the function "describe"), I first looked at the value counts for the categorical columns ("region" and "school type"). I then began grouping different columns (using the "groupby" function) to gain a deeper insight into the data provided. I calculated the median starting and midcareer salary for each school type in the cleaned dataset. I also calculated the mean salary for each region in the 10th and 90th percentile groups. To understand more indepthly the potential for income inequality, I substracted these values from each other. Lastly, using feature engineering I used a user defined function to calculate the economic classes based on the salaries (information I got from the Census' Bureau). From there I created a new column based on this information. Finally, I saved the final, cleaned and refined version of the dataset.

#### **Section 4: Observations and Questions**

***Takeaways/Anaylsis***: <br>
1. Ivy League schools lead in salary growth: Both the median starting salary and median mid-career salary are highest for the Ivy League category, followed by Engineering schools. Additionally, Ivy League graduates experience the largest salary increase over time, rising from 59,250 at the start of their careers to 115,000 at mid-career, highlighting a significant long-term earning advantage.
2. Wide salary disparities across and within regions: There is a striking gap between the 10th and 90th percentile mid-career salaries, both across different regions and within them. The Northeastern region has the highest salaries in both percentiles (48,610 at the 10th percentile and 180,985 at the 90th percentile), but it also shows the largest salary difference of 132,375. This highlights the significant income inequality within regions, where the highest earners make substantially more than those at the lower end of the spectrum.
<br>

***Questions***:
* Based on my takeaways above, two potential questions to continue exploring are:
    * How do starting and mid-career salaries vary across different types of colleges (State, Engineering, Ivy League, etc.)?
        * Could potentially show through a bar chart
    * Is there a regional disparity in career earnings among college graduates?
        * Could potentially show using a heatmap 

#### **Conclusion:**

Overall, this analysis highlights the significant impact that college institutions have on career earnings, with Ivy League and Engineering schools leading in both starting and mid-career salaries. Additionally, the substantial gap between the 10th and 90th percentile salaries, especially within regions like the Northeast, underscores the broader issue of income inequality among graduates. These findings suggest that while higher education can be a strong predictor of earning potential, systemic disparities in institutional resources and regional job markets play a crucial role in shaping long-term financial outcomes.